### PR TITLE
Converted "Download" to "Save" to local file system [#133108353]

### PR DESCRIPTION
**Pull request by dougmartin**
* Added file save to local file provider
* Fixed missing sharing code changes for v2 document store
* Added download/save when enter key is pressed in save and download dialogs
* Removed "Download" from default menu

Viewable here: https://codap.concord.org/~dmartin/add-local-file-save/static/dg/en/cert/index.html

### CODE REVIEW

File: src/assets/examples/auto-saving.html [hunk 1]
```diff
@@ -1,47 +1,48 @@
...
+        ui: {
+          menuBar: {
+            info: "This version auto saves every 5 seconds"
+          }
+        }
+      };
+      CloudFileManager.createFrame(options, "wrapper");
+    </script>
+  </body>
+</html>
```
Comment (kswenson) [inline src/assets/examples/auto-saving.html]: Why does Github think every line has changed here? Is/was there a line-ending issue?

File: src/code/client.coffee [hunk 2]
```diff
@@ -334,11 +334,15 @@ class CloudFileManagerClient
       @saveFileDialog stringContent, callback
 
   saveFile: (stringContent, metadata, callback = null) ->
-    if metadata?.provider?.can 'save', metadata
+    if metadata?.provider?.can('save') and not metadata.forceSaveDialog
```
Comment (kswenson) [inline src/code/client.coffee]: Shouldn't `metadata` be passed to the `can()` function here?

File: src/code/providers/document-store-share-provider.coffee [hunk 3]
```diff
@@ -84,7 +84,7 @@ class DocumentStoreShareProvider
         xhrFields:
           withCredentials: true
         success: (data) ->
-          # on successful share/save, capture the sharedDocumentId and shareEditKey
+          # on successful share/save, capture the sharedDocumentId and accesKeys
           masterContent.addMetadata
```
Comment (kswenson) [inline src/code/providers/document-store-share-provider.coffee]: nit: `accesKeys` => `accessKeys`

File: src/code/views/local-file-tab-save-view.coffee [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      filename: filename
+      trimmedFilename: @trim filename
+      shared: @props.client.isShared()
+      includeShareInfo: false
+      gotContent: false
+      content: null
+
+  componentDidMount: ->
+    @props.client._event 'getContent', { shared: @props.client._sharedMetadata() }, (content) =>
+      debugger
```
Comment (kswenson) [inline src/code/views/local-file-tab-save-view.coffee]: Remove the `debugger` statement? 😉

File: src/code/client.coffee (hunk 2 above)
Comment (dougmartin) [inline src/code/client.coffee]: I didn't see that you added the extra parameter in the lara provider and I thought it was a copy-paste error.  I'll add it back.

File: src/assets/examples/auto-saving.html (hunk 1 above)
Comment (dougmartin) [inline src/assets/examples/auto-saving.html]: It looks like it - VS Code likes to default to CRLF.  I'll change it back to LF.

File: src/code/providers/document-store-share-provider.coffee (hunk 3 above)
Comment (dougmartin) [inline src/code/providers/document-store-share-provider.coffee]: Fixed.

File: src/code/views/local-file-tab-save-view.coffee (hunk 4 above)
Comment (dougmartin) [inline src/code/views/local-file-tab-save-view.coffee]: Oops :)

**Pull request merged**